src/Microsoft.Diagnostics.Monitoring/Egress/FileSystem/FileSystemEgressProvider.cs

```diff
@@ -121,5 +124,91 @@ private async Task WriteFileAsync(Func<Stream, CancellationToken, Task> action,
             await fileStream.FlushAsync(token);
             Logger?.LogDebug("End writing to file.");
         }
+
+        private static void WrapException(Action action)
+        {
+            try
+            {
+                action();
```

Comment:
nit: just call something like WrapException( () => {action(); return true;}) instead of redefining the catch block.

Reply:
Done